go/enclave/components/batch_executor.go

```diff
@@ -203,9 +208,66 @@ func (executor *batchExecutor) ComputeBatch(ctx context.Context, context *BatchE
 		txReceipts = append(txReceipts, txResult.Receipt)
 	}
 
+	onBlockTx, err := executor.systemContracts.CreateOnBatchEndTransaction(ctx, stateDB, successfulTxs, txReceipts)
+	if err != nil && !errors.Is(err, system.ErrNoTransactions) {
+		return nil, fmt.Errorf("could not create on block end transaction. Cause: %w", err)
+	}
+	if onBlockTx != nil {
+		onBlockPricedTxes := common.L2PricedTransactions{
+			common.L2PricedTransaction{
+				Tx:             onBlockTx,
+				PublishingCost: big.NewInt(0),
+			},
+		}
+		onBlockSuccessfulTx, _, onBlockTxResult, err := executor.processTransactions(ctx, batch, len(successfulTxs), onBlockPricedTxes, stateDB, context.ChainConfig, true)
+		if err != nil {
+			// todo: remove once feature finished
```

Comment:
not sure what this is about

Reply:
There were panics on unexpected errors and the todo is for them